arviz/utils.py

```diff
@@ -100,3 +100,66 @@ def format_sig_figs(value, default=None):
     if value == 0:
         return 1
     return max(int(np.log10(np.abs(value))) + 1, default)
+
+
+def conditional_vect(function=None, **kwargs):  # noqa: D202
+    """Use numba's vectorize decorator if numba is installed.
+
+    Notes
+    -----
+        If called without arguments  then return wrapped function.
+        @conditional_vect
+        def my_func():
+            return
+        else called with arguments
+        @conditional_vect(nopython=True)
+        def my_func():
+            return
+
+    """
+
+    def wrapper(function):
+        try:
+            numba = importlib.import_module("numba")
+            return numba.vectorize(**kwargs)(function)
+
+        except ImportError:
+            return function
+
+    if function:
+        return wrapper(function)
+    else:
+        return wrapper
+
+
+def numba_check():
```

Comment:
Can this be added to the Numba class to consolidate all the logic?

It doesn't have to be but just asking for stylistic and organization reasons

Reply:
In my opinion, putting `numba_check` inside the class would not work as `numba_flag` is a class variable and calling class functions within a class is not possible in python. To get it working, I might have to switch to instance but that might lead to further breakdowns.

Reply:
In my opinion, putting `numba_check` inside the class would not work as `numba_flag` is a class variable and calling class functions within a class is not possible in python. To get it working, I might have to switch to instance but that might lead to further breakdowns.

Reply:
Can class functions not be called within a class? The following code works, one class method calls the other. Are we thinking of the same thing?

```
class Test:
    @classmethod
    def say_hi(cls): 
        print("hi")

    @classmethod 
    def say_hi_for_real(cls):
        print("Really gonna say hi")
        cls.say_hi()
    
Test.say_hi_for_real()
```

> Really gonna say hi
> hi

Reply:
My aim is to set `numba_flag` to `numba_check()` inside the class itself. 

    @classmethod
    def numba_check(cls):
        """Check if numba is installed."""
        numba = importlib.util.find_spec("numba")
        return numba is not None

    numba_flag = numba_check()

    @classmethod
    def disable_numba(cls):
        """To disable numba."""
        cls.numba_flag = False

    @classmethod
    def enable_numba(cls):
        """To enable numba."""
        if numba_check():
            cls.numba_flag = True
        else:
            raise ValueError("Numba is not installed")

Reply:
I was aiming for something like this:
class Numba:


    numba_flag = None

    @classmethod
    def numba_check(cls):
        """Check if numba is installed."""
        numba = importlib.util.find_spec("numba")
        return numba is not None

    numba_flag = Numba.numba_check()

    @classmethod
    def disable_numba(cls):
        """To disable numba."""
        cls.numba_flag = False

    @classmethod
    def enable_numba(cls):
        """To enable numba."""
        if cls.numba_check():
            cls.numba_flag = True
        else:
            raise ValueError("Numba is not installed")


A possible workaround is the following :

    @classmethod
    def numba_check(cls):
        """Check if numba is installed."""
        numba = importlib.util.find_spec("numba")
        return numba is not None

    @classmethod
    def set_flag(cls):
        cls.numba_flag = cls.numba_check()

    @classmethod
    def disable_numba(cls):
        """To disable numba."""
        cls.numba_flag = False

    @classmethod
    def enable_numba(cls):
        """To enable numba."""
        if cls.numba_check():
            cls.numba_flag = True
        else:
            raise ValueError("Numba is not installed")

But this would lead extra code and tests which would defeat the purpose of including `numba_check` inside the class. 
I guess leaving `numba_check` outside the class should be fine.

Reply:
Ah I see. Alright no worries was just curious